src/components/modal/modal.jsx

```diff
@@ -0,0 +1,195 @@
+  width: 40px;
+  height: 2px;
+  background: rgba(246, 246, 246, 0.5);
+  border-radius: 1px;
+  margin: 0 auto;
+  cursor: grab;
+  @media (min-width: 1000px) {
+    display: none;
+  }
+`;
+const ModalTitle = styled.h5`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 10px;
+  line-height: 120%;
+  text-align: center;
+  text-transform: uppercase;
+  font-feature-settings: "pnum" on, "lnum" on;
+  color: rgba(246, 246, 246, 0.7);
+  margin: 32px 0 16px;
+  @media (min-width: 1000px) {
+    margin: 84px 0 12px;
+  }
+`;
+const ModalNotificationBox = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const ModalText = styled.p`
+  font-style: normal;
+  font-weight: 700;
+  font-size: 18px;
+  line-height: 120%;
+  text-align: center;
+  font-feature-settings: "pnum" on, "lnum" on;
+  color: #f6f6f6;
+  margin-bottom: 32px;
+`;
+const ModalFooter = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-self: flex-end;
+`;
+const ModalCloseButton = styled.div`
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+  position: absolute;
+  top: 26px;
+  right: 26px;
+  width: 12px;
+  height: 12px;
+  background-color: transparent;
+  background-image: url("${closeIcon}");
+  background-position: center;
+  background-size: cover;
+  border: none;
+  cursor: pointer;
+  transition: opacity 0.3s;
+  @media (max-width: 1000px) {
+    display: none;
+  }
+`;
+
+export const Modal = ({ isOpen, onClose, title, text, children }) => {
+  const y = useMotionValue(0);
+  const isLessThan1000 = useMediaQuery("(max-width:1000px)");
+  useEffect(() => {
+    if (!isOpen) {
+      onClose();
+    }
+  }, [isOpen]);
+
+  return (
+    <AnimatePresence>
+      {" "}
+      {isOpen && (
+        <ModalComponent
+          className="modal"
+          role="modal"
+          initial={{y: isLessThan1000 ? "100%" :"", opacity: 0 }}
+          animate={{ y: isOpen ? 0 : "100%", opacity: 1 }}
+          exit={{ y: isLessThan1000 ?"100%":"", opacity: 0 }}
+          transition={{
+            duration: 0.3,
+            type: "spring",
+            damping: 50,
+            stiffness: 200,
+          }}
+          drag="y"
+          dragElastic={0.1}
+          dragMomentum={false}
+          dragConstraints={{ top: 0, bottom: 0 }}
+          onDragEnd={(event, info) => {
+            if (info.offset.y > 300) {
+              onClose();
+            }
+          }}
+        >
+          <ModalWrapper>
+            <ModalCloseButton
+              type="button"
+              onClick={onClose}
+              aria-label="Close modal"
+              data-bs-dismiss="modal"
+            ></ModalCloseButton>
+            <ModalBody>
+              <ModalHeader />
+              <ModalTitle>{title}</ModalTitle>
+              <ModalNotificationBox>
+                {children}
+                <ModalText>{text}</ModalText>
+              </ModalNotificationBox>
+            </ModalBody>
+            <ModalFooter>{children}</ModalFooter>
```

Comment:
Should you go ahead and use the Button Component to make the "Sign" and "Decline" buttons and put them here instead of children? Modal would also need props like onSignClick / onDeclineClick to pass the handler functions down to the button

Reply:
To expand here-- I think it's appropriate to have `{children}` here because this modal will be re-used for a lot of different purposes and not all of the modals have those buttons, so they should be passed in by the consumer. With that in mind, those functions for `onSign` and `onDecline` would be click listeners on those buttons and wouldn't need to be passed into the modal.

However, I think there's some confusion here about how `{children}` works; there are multiple places where you're using it, such as lines 169 and 173, but `children` refers to the same thing in both instances. What I would prefer is that the entire `ModalBody` here be replaced with `{children}` and then when the consumer uses this component, you will pass in the children like so:
```
<Modal>
   <Modal.Header>...</Modal.Header>
   <Modal.Body>...</Modal.Body>
   <Modal.Footer>...</Modal.Footer>
</Modal>
```
or
```
<Modal>
   <Modal.Header>...</Modal.Header>
   <Modal.Body>...</Modal.Body>
</Modal>
```

That way, some modals may have footers while others don't, etc.